src/renderer/effectTransformPoint.js

```diff
@@ -0,0 +1,77 @@
+import { effectBitmasks } from "./effectInfo.js";
+
+const CENTER = 0.5;
+const EPSILON = 1e-3;
+
+// Transform a texture-space point using the effects defined on the given drawable.
+const effectTransformPoint = (drawable, src, dst) => {
```

Comment:
Simulating effect transformations in JavaScript is really interesting to me, even though the details are a bit beyond my comprehension :P

I think it'd be interesting to look over the discussion/summary in LLK/scratch-render#196 and try testing [the example project](https://scratch.mit.edu/projects/130041250/) in Leopard with these changes and "if on edge, bounce" implemented.